go/login/login_test.go

```diff
@@ -0,0 +1,181 @@
+			url:        "/v1/AYUg8AmJMKdUdIt93LQ-91oNvzoNJjga9OukqY6qm05q0PU=/my-server.local/shell/root/",
+			prefixN:    decodeString("d0f59d0b17cb155a1b9cd2b5cdea3a17f37a200e95e3651af2c88e1c5fc8108e"),
+			tag:        decodeString("9c44389f462d35d0672faf73a5e118f8b9f5c340bbe8d340e2b947c205ea4fa3"),
+			token:      "lyHuaHuCck",
+		},
+
+		{
+			kap:        decodeString("b105f00db105f00db105f00db105f00db105f00db105f00db105f00db105f00d"),
+			ka:         decodeString("d1b6941bba120bcd131f335da15778d9c68dadd398ae61cf8e7d94484ee65647"),
+			kbp:        decodeString("fee1deadfee1deadfee1deadfee1deadfee1deadfee1deadfee1deadfee1dead"),
+			kb:         decodeString("872f435bb8b89d0e3ad62aa2e511074ee195e1c39ef6a88001418be656e3c376"),
+			ks:         decodeString("4b1ee05fcd2ae53ebe4c9ec94915cb057109389a2aa415f26986bddebf379d67"),
+			prefix:     byte(prefix2),
+			hostIdType: "serial-number",
+			hostId:     "1234567890=ABCDFGH/#?",
+			action:     "reboot",
+			msg:        []byte("serial-number:1234567890=ABCDFGH/#?/reboot"),
+			url:        "/v1/UYcvQ1u4uJ0OOtYqouURB07hleHDnvaogAFBi-ZW48N2/serial-number:1234567890=ABCDFGH%2F%23%3F/reboot/",
+			prefixN:    decodeString("dff5aae753a8bdce06038a20adcdb26c7be19cb6bd05a7850fae542f4af29720"),
+			tag:        decodeString("06476f1f314b06c7f96e5dc62b2308268cbdb6140aefeeb55940731863032277"),
+			token:      "p8M_BUKj7zXBVM2JlQhNYFxs4J-DzxRAps83ZaNDquY=",
+		},
+	}
+}
+
+func ClientsAndServers(t *testing.T) ([]Client, []Server) {
+	tvs := testVectors()
+	var keyPairs [tvsCnt][]KeyPair
+	for i, tv := range tvs {
+		keyPairs[i] = append(keyPairs[i], *keys(t, tv.kap, tv.ka), *keys(t, tv.kbp, tv.kb))
+	}
+
+	var clients []Client
+	var servers []Server
+	for tv := 0; tv < tvsCnt; tv++ {
+		clients = append(clients, *NewClient(keyPairs[tv][1].pub, keyPairs[tv][0].priv, serviceKeyIds[tv]))
+		sPrivKey := keyPairs[tv][1].priv
+		servers = append(servers,
+			Server{func(u uint8) (glome.PrivateKey, error) {
+				return sPrivKey, nil
+			}})
+	}
+	return clients, servers
+}
+
+func TestCheckCorrectURLs(t *testing.T) {
+	_, servers := ClientsAndServers(t)
+	var responses []URLResponse
+	tvs := testVectors()
+
+	for i, tv := range tvs {
+		t.Run("Test vector "+fmt.Sprint(i+1), func(t *testing.T) {
+			resp, err := servers[i].ParseURLResponse(tv.url)
+			if err != nil {
+				t.Fatalf("TestCheckCorrectURLs failed for test vector %d. Expected: parsed URL, got: %v", i, err)
+			}
+
+			responses = append(responses, *resp)
+		})
+	}
+
+	for i, tv := range tvs {
+		t.Run("Test vector "+fmt.Sprint(i+1), func(t *testing.T) {
+			for _, k := range []struct {
+				expected string
+				got      string
+			}{
+				{expected: tv.hostIdType, got: responses[i].Msg.HostIDType},
+				{expected: tv.hostId, got: responses[i].Msg.HostID},
+				{expected: tv.action, got: responses[i].Msg.Action},
+			} {
+				if k.expected != k.got {
+					t.Fatalf("TestCheckCorrectURLs failed for test vector %d. Expected: %v, got: %v", i, k.expected, k.got)
```

Comment:
How does the error actually look? `%#+v` might potentially be a better string format for expected and got.